service/matching/version_sets.go

```diff
@@ -30,13 +30,25 @@ import (
 	"crypto/sha256"
 	"encoding/base64"
 
+	commonpb "go.temporal.io/api/common/v1"
 	"go.temporal.io/api/serviceerror"
 	"go.temporal.io/api/taskqueue/v1"
 	"go.temporal.io/api/workflowservice/v1"
 	"go.temporal.io/server/api/persistence/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
 	hlc "go.temporal.io/server/common/clock/hybrid_logical_clock"
 )
 
+var (
+	// TODO: go over error types, maybe not all should be invalid argument
+	errBuildNotFound                     = serviceerror.NewInvalidArgument("build id not found")
+	errNewerBuildFound                   = serviceerror.NewInvalidArgument("newer compatible build exists")
+	errEmptyVersioningData               = serviceerror.NewInvalidArgument("versioning data is empty")
+	errPollWithVersionOnUnversionedQueue = serviceerror.NewInvalidArgument("poll with version capabilities on unversioned queue")
```

Comment:
Hmm... this will be bad experience for users if they deploy their versioned workers before updating the versioning data.
Same as above (worker crashes in Core).

Reply:
See my comment below how the fact that there's no versioning data doesn't necessarily imply that a queue is unversioned.

Reply:
I'll leave TODOs for these since it's not core functionality, it's only for failed replication and similar situations